wolfcrypt/test/test.c

```diff
@@ -6313,8 +6434,28 @@ WOLFSSL_TEST_SUBROUTINE wc_test_ret_t hash_test(void)
         if (exp_ret == 0 && hashType != typesGood[i])
             ERROR_OUT(WC_TEST_RET_ENC_I(i), out);
 #endif /* !defined(NO_ASN) || !defined(NO_DH) || defined(HAVE_ECC) */
-    }
 
+#if defined(WOLFSSL_SMALL_STACK) && !defined(WOLFSSL_NO_MALLOC)
+        if (exp_ret == WC_NO_ERR_TRACE(HASH_TYPE_E)) {
```

Comment:
I don't like this still. Why re-init a failed init just to delete it? Can't you just skip the hash delete in that case? If you can't then there is a different issue you should fix to make sure a partial init failure actually free's the resource.

Reply:
I missed this one. You are correct: now that I've removed the placeholder hash type, there's no reason to do this. Good catch.